ionc/ion_extractor.c

```diff
@@ -220,6 +222,9 @@ iERR ion_extractor_path_create_from_ion(ION_EXTRACTOR *extractor, ION_EXTRACTOR_
     }
     IONCHECK(ion_reader_step_in(reader));
     for (;;) {
+        if (path_length >= extractor->_options.max_path_length) {
+            FAILWITHMSG(IERR_INVALID_ARG, "Illegal number of path components.");
```

Comment:
I might make the message indicate the number given and the maximum exceeded to be a bit more informative.

Reply:
Yes, or at least change it to say "Maximum number of path components exceeded" or somesuch.